app/controllers/nsm/steps/disbursement_cost_controller.rb

```diff
@@ -29,7 +32,24 @@ def ensure_disbursement
       end
 
       def additional_permitted_params
-        [:apply_vat]
+        [:apply_vat, :add_another]
+      end
+
+      def new_record?
+        disbursement.disbursement_date.nil? || disbursement.miles.nil?
+      end
+
+      def build_flash
+        if new_record?
+          if params.dig(:nsm_steps_disbursement_cost_form, :add_another) == YesNoAnswer::YES.to_s
+            count = current_application.disbursements.count + 1
```

Comment:
This won't work. When you're adding your first disbursement, by this point it already exists in the database so the `current_application.disbursements.count` is 1. Adding 1 to it gets you 2, so on the next screen you'll see "You've added 2 disbursements" when in fact you've only added 1.